whois-commons/src/main/java/net/ripe/db/whois/common/rpsl/transform/FilterAuthFunction.java

```diff
@@ -172,4 +184,43 @@ private boolean passwordAuthentication(final List<RpslAttribute> authAttributes)
         }
         return false;
     }
+
+    private boolean apiKeyAuthenticated(final RpslObject rpslObject, final List<RpslObject> maintainers, final List<RpslAttribute> authAttributes) {
+        if(oAuthSession == null || oAuthSession.getUuid() == null) {
+            return false;
+        }
+
+        if(rpslObject.getType().equals(ObjectType.MNTNER)) {
+            maintainers.add(rpslObject);
+        }
+
+        if(!oAuthSession.getScopes().isEmpty()) {
+            final OAuthSession.ScopeFormatter scopeFormatter = new OAuthSession.ScopeFormatter(oAuthSession.getScopes().getFirst());
+            if(!validateScope(maintainers, scopeFormatter)) {
+                return false;
+            }
+        }
+
+        for (final RpslAttribute attribute : authAttributes) {
+            final Matcher matcher = SSO_PATTERN.matcher(attribute.getCleanValue().toString());
+            if (matcher.matches()) {
+                try {
+                    if (oAuthSession.getUuid().equals(matcher.group(1))) {
+                        return true;
+                    }
+                } catch (AuthServiceClientException e) {
+                    return false;
+                }
+            }
+        }
+
+        return false;
+    }
+
+    private static boolean validateScope(final List<RpslObject> maintainers, final OAuthSession.ScopeFormatter scopeFormatter) {
+        return scopeFormatter.getAppName().equalsIgnoreCase("whois")
```

Comment:
are we ignoring the fourth part of the scope?
It will always be "write" right now.
Maybe we should check it...just for being consistent?
I don't know, I leave it to you

Reply:
Yes , it will not work as I need to add the dependency whois-update on whois-query which I don't want and this does not belong to whois-common.I am not sure, I can move the whole APIKeyValidator to whois common but that will also not work as there are lot of implementation of that interface so need to do lot of refactoring. We are doing the same thing ClientAuthCertificateValidator so I just did the same for consitency.

Reply:
May be I can do other way around. Let me try

Reply:
Thanks for the change